### ISSUE TITLE
ArC does not permit the use of 3rd Party CDI libraries

### ISSUE THREAD

**Issue by dansiviter** [quarkusio/quarkus#2274]
According to the [CDI Reference#Limitations](https://quarkus.io/guides/cdi-reference.html#limitations):
> Portable Extensions are not supported

Therefore, it is not possible to use CDI libraries such as Apache DeltaSpike that utilise the `javax.enterprise.inject.spi.Extension` mechanism.

Although this is documented (via #559), not implementing the CDI specification causes integration and migration headaches as it requires re-implementing via `BeanRegistrarBuildItem`.

Is the eventual goal that ArC will cover the complete CDI specification?

**Comment by mkouba** [comment]
> Is the eventual goal that ArC will cover the complete CDI specification?

No, it's not a goal to implement all CDI features and pass the CDI TCK. CDI portable extensions are indeed very powerful yet they do not match the Quarkus "design philosophy" (build time metadata processing, reduction in reflection usage, etc.). We would like to provide the **build time extension points** for the most common tasks that are usually solved by CDI  portable extensions.

**Comment by dansiviter** [comment]
This cherry-picking policy is causing more problems than it solves.
